### ISSUE TITLE
XML output with findbugs style should have -showStats set implicitly

### ISSUE THREAD

**Issue by TheRealAgentK** [cflint/CFLint#300]
Findbugs style output makes no sense whatsoever without the -showStats flag now that we're following the proper Findbugs.xsd.

We should consider switching it on implicitly for -xmlstyle findbugs even if it's not being passed in.

Any opinions @ryaneberly, @denuno ?

**Comment by ryaneberly** [comment]
Interesting.  I agree.   I think maybe I thought showStats would drive the
printout on the console.  Collecting them doesn't cost much..  I'm ok with
assumed they are always on.

**Comment by TheRealAgentK** [comment]
Ok - so, I looked into the output flow dependencies a bit to understand this a bit better.

A) XML: 

1. CFLint XML Format gets created anyway
2. If style == findbugs -> XSLT gets applied to CFLint XML and then written, otherwise CFLint XML Format gets written.

In the XML case showStats controls the output of the counts-XML-element into the CFLint output. As this is the basis for Findbugs XSTL, if CFLint won't write it, it won't be in Findbugs.

B) HTML:

1. CFLint XML Format gets created anyway
2. HTML style default or plain gets applied by XSLT transformation again. 

Here, the HTML style transformation for the templates needs the stats anyway, otherwise it plainly can't fill the template properly. Same as in A - if CFLint XML doesn't have it, it won't be in HTML.

C) JSON:

JSON is a simpler process - JSON gets written out ion JSONOutput directly. If showStats is off, the objects for counts-by-code and -severity simply won't be in the output (it will however still have been collected regardless).

D) TEXT:

Similar to JSON, simple process. If showStats is off outputs for "Issues counts" and "Total issues" just won't happen (again it will have been collected).


TL;DR;

- We're collecting all the bug count information anyway when CFLint runs. showStats=false does not provide any performance gain whatsoever.
- XML Findbugs needs showStats=true to be useful and HTML needs  showStats=true to be useful, we might as well enable it by default for all CFLint XML.

The question then is: Should we enable it for all output in CFLint or only for XML-based output?

Is there any use case we could think of why someone would NOT want to have the counts for JSON or TEXT output? I struggle to see why having them in there would be an issue, frankly. If we removed the showStats completely , it'd be a much cleaner solution in my opinion (as opposed to having a showStats=true default for XML and a showStats=false default for TEXT and JSON)

**Comment by ryaneberly** [comment]
+1.  on removing the showStats option entirely.

**Comment by TheRealAgentK** [comment]
We now always create the counts section, the stats flag is gone and showStats function parameters have all been refactored.
